### ISSUE TITLE
CNI Support Cross-Subnet Pod Communication

### ISSUE THREAD

**Issue by WLBF** [kubeedge/edgemesh#278]
**What would you like to be added/modified**:
Support cross-subnet pod communication

**Why is this needed**:
It's common that cloud node can not reach edge node behind NAT. So It is a very important capability to allow pods to communicate across subnets. For example, prometheus stack adapt scrape model to pull data from exporters running on the nodes, It's barely possible using prometheus stack mointering edge cluster if prometheus on cloud can not access pod runinng on the edge node.

**Comment by Poorunga** [comment]
Yes, edgemesh support for pod ip communication is planned. However, the exact plan is still very much up for discussion. 

**what problem do we need to solve**:
- edgemesh supports cni plugin(such as flannel, calico). This ensures that each pod has a unique IP address
- edgemesh supports cross-subnet traffic transport at the pod IP level

However, if the pod uses hostNetwork mode, the problem is more difficult to solve. At this time, we need to solve the problem similar to VPN.
